source/AI.cpp

```diff
@@ -1978,7 +1978,12 @@ void AI::PrepareForHyperspace(Ship &ship, Command &command)
 	bool isJump = (ship.GetCheapestJumpType(ship.GetTargetSystem()).first == Ship::JumpType::JumpDrive);
 
 	Point direction = ship.GetTargetSystem()->Position() - ship.GetSystem()->Position();
-	if(!isJump && scramThreshold)
+	if(ship.Position().Length() <= ship.GetSystem()->DepartureDistance())
+	{
+		Point closestDeparturePoint = ship.Position() * (ship.GetSystem()->DepartureDistance() / ship.Position().Length()) * 1.1;
```

Comment:
Does this also work if the Ship is at position (0,0)? And would it make sense to add a (named) constant instead of multiplying with 1.1 to ensure we get outside the range? Very large departure distances would give quite some overshoot with a factor of 1.1.

Reply:
yeah, this was for testing, I can remove it now

Reply:
I'm afraid we still have a division by zero if the Ship is at position (0,0), because I would expect both Length() and LengthSquared() to return 0 in such cases.
And even if we would not have a division by zero, the calculation of the closestDeparturePoint would still result in position(0,0), because multiplying zero with anything doesn't give any new point.

Reply:
fixed